CONTRIBUTING.md

```diff
@@ -0,0 +1,77 @@
+
+### A brief summary of CC BY 4.0 License
+- Your work can be distributed so long as you authorship remains attached (known as Attribution)
+- Your work is not exempt from Commerical use
+- Your work may be modified by another person, as long as your original authorship is retained on the work (known as Derivative Works)
+- You are free of any liability or warranty for your works
+
+### Including works by other authors (aka redistribution and derivative works)
+CC BY 4.0 licensed files allow for redistribution providing you attribute the author. Derivative works must contain the original author and the license used; and your own authorship. Other licenses used may include the MIT License (used in CorsixTH's base code), which allow for reuse/modification again with reference to original author. **However, it is your responsibility to ensure all included files have compliant licenses for your use.**
+
+### License file
+The opening of your license must contain a reference to yourself as author with the year the works were created. It must also include the name of your level/map/campaign as appropriate.
+```
+CONTENT NAME level/map/campaign is freely distributed under the following terms:
+Copyright (C) YEAR FORENAME "USERNAME" SURNAME
+```
+where:
+- `CONTENT NAME` is the name of your work
+- `level/map/campaign` delete as appropriate
+- `YEAR` the year of creation
+- `FORENAME "USERNAME" SURNAME` Forename and Surname are at your own discretion. You may use whatever identifier you like as long as it is a representation of you.
+- If your submission is a derivative work, place the original author (and year) before your own authorship if the original work used CC BY 4.0 too e.g. `Copyright (C) 2020 Joe "jbloggy" Bloggs, 2022 John "TrekkingJim" Doe`. If not, see redistributed works section below.
+
+The license file must contain for your work a link out to the CC BY 4.0 license. Luckily, Creative Commons' required text is very small.
+```
+This work is licensed under a Creative Commons Attribution 4.0 International License - https://creativecommons.org/licenses/by/4.0/
+```
+If you're **redistrubiting someone elses work**, you must proceed to declare this under your license using the original author's license and any stipulations given. This includes mentioning the name of what level/maps of your work are by someone else. 
+**Tip:** you can view the [MIT License](https://choosealicense.com/licenses/mit/) text here.
+
+### License in level files
+You must also follow the practice above again on a per-file basis for `.level` and `.campaign` files. `.map` files can't currently have authorship appended and rely on the `LICENSE.txt`. Note, authorship here is file specific. Such as, if one file is solely your work and a different file is both yours and someone else's work: you must apply the licensing accordingly.
```

Comment:
Actually only map files can't, both .level and .campaigns have comments that can be used to supply authorship and licence information - but sure a LICENSE.txt file.

Reply:
I think in future we should make the Map Editor have a section added for license declaration (or metadata?).
Or @Alberth289346 suggested converting the .map to a text file with instructions on how to build the map.

Reply:
Actually, I intended to suggest an alternative file format for maps, ie in text. Perhaps something with rectangles of tiles that have some sprite?

Something like
```
rect (1, 4)-(50,8): dark-green grass;
line (20, 50)-(20-80): west outer-wall;
```
fills the rectangle between (and inclusive) between (1,4) and (50, 8) with a sprite named "dark-green grass", and places a west outer-wall between and inclusive (20, 50) and (20, 80).

The format can have more fields for storing other information of course, alternative ways to express sprites (eg number rather than name), etc etc.

Reply:
I would say I like the concept in theory. Let's move it to a new discussion outside of here though as I imagine we'd be a way off transitioning right now.